bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/EntryLogManagerForEntryLogPerLedger.java

```diff
@@ -93,7 +93,7 @@ BufferedLogChannel getLogChannel() {
         private BufferedLogChannelWithDirInfo entryLogWithDirInfo;
 
         private EntryLogAndLockTuple(long ledgerId) {
-            int lockIndex = Long.hashCode(ledgerId) % lockArrayPool.length();
+            int lockIndex = Math.abs(Long.hashCode(ledgerId) % lockArrayPool.length());
```

Comment:
If `hashCode()` returns a negative, the modulo operation will fail anyway. I think we should use `MathUtils.signSafeMod()` here.

Reply:
https://docs.oracle.com/javase/specs/jls/se7/html/jls-15.html#jls-15.17.3

in java for modulo (remainder) operation both right and left operands can be negative.

Reply:
Sure, though I'd prefer to consistently use `signSafeMod()` like in rest of codebase

Reply:
changed it to signSafeMod